expression/builtin_json_vec.go

```diff
@@ -847,11 +848,87 @@ func (b *builtinJSONMergeSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Colu
 }
 
 func (b *builtinJSONContainsPathSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONContainsPathSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	jsonBuf, err := b.bufAllocator.get(types.ETJson, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(jsonBuf)
+	if err := b.args[0].VecEvalJSON(b.ctx, input, jsonBuf); err != nil {
+		return err
+	}
+	typeBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(typeBuf)
+	if err := b.args[1].VecEvalString(b.ctx, input, typeBuf); err != nil {
+		return err
+	}
+	pathBufs := make([]*chunk.Column, len(b.args)-2)
+	defer func() {
+		for i := 0; i < len(pathBufs); i++ {
+			if pathBufs[i] != nil {
+				b.bufAllocator.put(pathBufs[i])
+			}
+		}
+	}()
+	for i := 0; i < len(pathBufs); i++ {
+		pathBuf, err := b.bufAllocator.get(types.ETString, n)
+		if err != nil {
+			return err
+		}
+		if err := b.args[2+i].VecEvalString(b.ctx, input, pathBuf); err != nil {
+			return err
```

Comment:
If the function return in this line, will `pathBuf` be garbage collected by bufAllocator ?

Reply:
```
defer func() {
		for i := 0; i < len(pathBufs); i++ {
			if pathBufs[i] != nil {
				b.bufAllocator.put(pathBufs[i])
			}
		}
	}()
```

I think it will collect garbage if function return in this line.

Reply:
But you set `pathBufs[i] = pathBuf` in L888 .  Move the line to L885?

Reply:
Thank you, fixed..